### ISSUE TITLE
Give better classes for VirtualMachineError

### ISSUE THREAD

**Issue by fubuloubu** [iamdefinitelyahuman/brownie#330]
### Overview
It should be possible to parse the error log from various EVM implementations and obtain the exception type from the logs (e.g. Invalid Instruction, Out of Gas, Invalid Jump Dest, etc.)

This will make it a lot easier to determine the type of exception, which could be used to make decisions on what to do about them.

**Comment by iamdefinitelyahuman** [comment]
To be clear, you're proposing to create new exceptions such as `EVMRevert` or `EVMInvalidInstruction`, based on the type of error?

**Comment by fubuloubu** [comment]
Yes, maybe subclassing the EVM errors under the current one so they can be differentiated but still share a common type

**Comment by iamdefinitelyahuman** [comment]
Useful reference for implementation: https://github.com/ethereum/py-evm/blob/master/eth/exceptions.py

All of the exceptions that subclass `VMError` could be replicated.
